packages/next/build/webpack-config.ts

```diff
@@ -1529,6 +1530,13 @@ export default async function getBaseWebpackConfig(
 
   const webpack5Config = webpackConfig as webpack5.Configuration
 
+  webpack5Config.module!.rules!.unshift({
+    test: /\.wasm$/,
+    issuerLayer: 'middleware',
+    loader: 'next-middleware-wasm-loader',
+    type: 'javascript/auto',
+  })
```

Comment:
This should be reiterated when (and if) `config.experimental.asyncWebAssembly` will be enabled by default, as we should probably use the default behavior with a custom runtime 🙏